cpp/src/barretenberg/stdlib/recursion/verification_key/verification_key.hpp

```diff
@@ -188,7 +188,7 @@ template <typename Curve> struct verification_key {
     }
 
   public:
-    field_t<Composer> compress()
+    field_t<Composer> compress(size_t const hash_index = 0)
     {
         field_t<Composer> compressed_domain = domain.compress();
 
```

Comment:
Need to do `key_witnesses.push_back(Composer::type)` here - I forgot to add it (see the `compress_native` static method directly below, for reference)

Reply:
Also might be worth renaming `key_witnesses` to `preimage_data`